### PR TITLE
Map displays mocked bike data using Axios

### PR DESCRIPTION
**Pull request by jonramm**
- Encapsulate Map as separate component
- Create BikeMarker component
- Set up Axios for HTTP requests in client app. Using axios-mock-adapter to provide app with test data.
- Create file for declaring types and keeping code DRY.
- Create 'services' folder in client app to store data fetching logic and functions. getBikes() returns array of mock bike objects.

### CODE REVIEW

File: client/components/Map.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            .catch(err => console.log(err));
+    }, [])
+
+    return (
+        <MapView 
+            style={styles.map}
+            region={{
+                latitude: props.latitude,
+                longitude: props.longitude,
+                latitudeDelta: 0.0922,
```
Comment (schaefkr) [inline client/components/Map.tsx:32]: What does latitudeDelta represent?

Comment (schaefkr) [inline client/components/Map.tsx:32]: What do _latitudeDelta_ and _longitudeDelta_ represent and why these values? Or are these just temporarily hardcoded?

File: client/configs/firebase.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+firebase.initializeApp(firebaseConfig);
+
+const auth = firebase.auth();
+// connectAuthEmulator(auth, "http://127.0.0.1:9099"); // emulator not working, defaulting to prod
+
+const provider = new firebase.auth.GoogleAuthProvider();
+provider.setCustomParameters({ prompt: 'select_account' });
+
+const signInWithGoogle = (provider: firebase.auth.AuthProvider) =>
```
Comment (schaefkr) [inline client/configs/firebase.ts:26]: Does this overlap with Anita's pull-request? https://github.com/jonramm/Bike_Kollective/pull/11

File: client/pages/Home.tsx [hunk 3]
```diff
@@ -1,7 +1,19 @@
 import React from "react";
-import { View, Text, Button, StyleSheet, Pressable } from 'react-native';
+import { View, Text, Button, StyleSheet, Pressable, TouchableOpacity } from 'react-native';
+import { useNavigation } from '@react-navigation/core'
+
+import { auth } from "../configs/firebase";
 
 const Home = ({navigation}) => {
+    // const navigation = useNavigation();
```
Comment (schaefkr) [inline client/pages/Home.tsx:8]: Does this overlap with Anita's pull-request? https://github.com/jonramm/Bike_Kollective/pull/11

File: client/pages/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
+import React, {useState, useEffect} from 'react';
+import { KeyboardAvoidingView, StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
+import { useTheme } from '@react-navigation/native';
+import { useNavigation } from '@react-navigation/core';
+
+import { firebase, auth, provider, signInWithGoogle } from '../configs/firebase';
+
+const styles = StyleSheet.create({
```
Comment (schaefkr) [inline client/pages/Login.tsx:8]: I'd prefer to have styles at the bottom of the doc, but it's not that important. Would be nice to have it consistent.

File: client/pages/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    useEffect(() => {
+        const login = auth.onAuthStateChanged(user => {
+            if (user) {
+                navigation.replace("Home")
+            }
+        });
+        return login;
+    }, []);
+
+    const handleSignUp = () => {
```
Comment (schaefkr) [inline client/pages/Login.tsx:69]: Again, I think there's overlap here. https://github.com/jonramm/Bike_Kollective/pull/11

File: functions/src/apis/bikes.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
+import {Response as responseType} from "express";
```
Comment (schaefkr) [inline functions/src/apis/bikes.ts:1]: I would prefer that in the future we merge one pull-request at a time: https://github.com/jonramm/Bike_Kollective/pull/10

**Review by schaefkr — COMMENTED** [review]
I will not approve for now, so that Anita has a chance to review as well.

File: client/components/Map.tsx (hunk 1 above)
Comment (jonramm) [reply client/components/Map.tsx:32]: According to [apple.developer.com](https://developer.apple.com/documentation/mapkit/mkcoordinatespan/1452573-longitudedelta) those properties represent the amount of east-to-west and north-to-south distance to display on the map. The values in there now were part of some example code I found so we can definitely fiddle with them.

File: client/configs/firebase.ts (hunk 2 above)
Comment (jonramm) [reply client/configs/firebase.ts:26]: Yeah, I've been starting my branches with other not-merged-yet branches. Maybe we could talk about best practices for that in our meeting?

**Pull request merged**